### DISCUSSION TITLE
incorrect intro music when selecting missing beatmap

### DISCUSSION THREAD

**Discussion by papertek** [ppy/osu#22736] [Q&A] [answered]
expected behavior is to have "welcome to osu!" play instead of "triangles!"

https://user-images.githubusercontent.com/72106503/221351647-80b73b04-9822-41a3-80ef-e22e1c1835af.mp4

**Comment by Axilotl17** [comment]
issue here is that the selected intro sequence does not play when the music isn't found; the selected intro sequence is ignored and the triangles intro (default intro sequence) is played. is this intentional or an oversight?

**Comment by bdach** [comment]
This is happening because the triangles intro imports itself as a fake beatmap so that its files are available to the game:

https://github.com/ppy/osu/blob/4e9d25957a2b9bf532a3ed1f1146d5c58c4f5d8a/osu.Game/Screens/Menu/IntroScreen.cs#L148-L150

but then the fake beatmap's audio is reachable via the `MusicController.EnsurePlayingSomething()` flow:

https://github.com/ppy/osu/blob/4e9d25957a2b9bf532a3ed1f1146d5c58c4f5d8a/osu.Game/Overlays/MusicController.cs#L230-L250

I'd argue that this is a mistake, and that "triangles" shouldn't be reachable via the music controller ever if it's not an imported beatmap, and it's better to play nothing in this case than do this. But I guess that is my opinion, mostly.

**Comment by Joehuu** [comment, marked as answer]
Related to https://github.com/ppy/osu/issues/21064.

**Reply by peppy** [reply]
I think we can continue to track this via this issue.
